### PR TITLE
add makeCalculator solution

### PR DESCRIPTION
**Pull request by DidenkoDenis306**

### CODE REVIEW

**Review by ruslanpashkov — APPROVED** [review]

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/makeCalculator.js [hunk 1]
```diff
@@ -37,7 +37,39 @@
...
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(num1, num2) {
+      return num1 + num2;
+    },
+
+    subtract(num1, num2) {
```
Comment (33Orange) [inline src/makeCalculator.js:47]: The subtract function should only take one parameter and subtract it from the current result.

File: src/makeCalculator.js [hunk 2]
```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(num1, num2) {
```
Comment (33Orange) [inline src/makeCalculator.js:43]: The add function should only take one parameter and add it to the current result.

File: src/makeCalculator.js [hunk 3]
```diff
@@ -37,7 +37,39 @@
...
+
+    subtract(num1, num2) {
+      return num1 - num2;
+    },
+
+    multiply(num1, num2) {
+      return num1 * num2;
+    },
+
+    divide(num1, num2) {
```
Comment (33Orange) [inline src/makeCalculator.js:55]: The divide function should only take one parameter and divide the current result by it.

File: src/makeCalculator.js [hunk 4]
```diff
@@ -37,7 +37,39 @@
...
+
+    add(num1, num2) {
+      return num1 + num2;
+    },
+
+    subtract(num1, num2) {
+      return num1 - num2;
+    },
+
+    multiply(num1, num2) {
```
Comment (33Orange) [inline src/makeCalculator.js:51]: The multiply function should only take one parameter and multiply the current result with it.

File: src/makeCalculator.js [hunk 5]
```diff
@@ -37,7 +37,39 @@
...
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(operation, n) {
+      this.result = operation(this.result, n);
```
Comment (33Orange) [inline src/makeCalculator.js:66]: Use `operation.call(context, argument)` to call function with new context and argument.

**Review by 33Orange — CHANGES_REQUESTED** [review]
A few moments:
- Update calculate methods. Functions should only take one parameter and add it to the current result.
- When you update methods use `.call()` to call your callback with provided context and arguments.

**Review by dinaminahi — CHANGES_REQUESTED** [review]
👍

**Review by naziamoff — APPROVED** [review]
🔥

**Pull request closed without merging**